substrate/node/src/chain_spec.rs

```diff
@@ -17,22 +14,22 @@ fn insecure_pair_from_name(name: &'static str) -> Pair {
   Pair::from_string(&format!("//{name}"), None).unwrap()
 }
 
-fn account_id_from_name(name: &'static str) -> AccountId {
+fn address_from_name(name: &'static str) -> NativeAddress {
```

Comment:
why did we switch from accountId to "NativeAddress"?

Reply:
We needed a primitive type for native addresses in serai-primitives. I added one which was equivalent to AccountId. AccountId didn't work as a name in the context of s-p. I then updated the runtime to re-export the primitives' type, instead of defining its own, to achieve a singular source of truth. While I could've had the runtime do `type AccountId = NativeAddress;`, I felt also have a singular name would be best. Given we're a cross-chain DEX, there is no "AccountId". Just the ID for an account on a chain. That's why specifying "Native" (or "Serai"), even within the Substrate code, is appreciated.

Reply:
The less we break from convention the better, this is a big one IMO. We should call this SeraiAddress, native is confusing.

Reply:
Yes, will call it SeraiAddress. Are we cool not then re-exporting SeraiAddress as AccountId?

The main issue is when we use this in the processor/coordinator, and that sees AccountId due to grabbing Substrate integration code, yet it's using it in a multi-chain context. That's why I put this naming even into Substrate itself.

Reply:
I vote for different types for external/native addresses. Not sure how we will use this in the processor/coordinator.

Reply:
We have that :+1: